bakery/mgostorage/storage.go

```
@@ -0,0 +1,96 @@
+// Package mgostorage provides an implementation of the
+// bakery Storage interface that uses MongoDB to store
+// items.
+package mgostorage
+
+import (
+	"github.com/juju/errgo"
+	"gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
+
+	"github.com/go-macaroon-bakery/macaroon-bakery/bakery"
+)
+
+// New returns an implementation of Storage
+// that stores all items in MongoDB.
+func New(c *mgo.Collection) *mgoStorage {
+	m := mgoStorage{
+		col: c,
+	}
+	m.setupCollection()
```

Now setupCollection is being called, we'll need to return an error from New,
otherwise any error is always ignored.